Docs/feature_memory_protection.md

```diff
@@ -1,359 +1,210 @@
-[MSR](#model-specific-register-msr),
-and then set the [XD](#no-executeexecute-neverexecute-disable-attribute-nxxnxd) bit in the CPU
-[PAE](#physicalpage-address-extension)
-page table.
-
-### Overhead
-**O(n)** time where n is the number of memory mapped regions. The number of actual set bits beyond one is
-inconsequential because every memory region must be checked if at least one bit is set. There is no extra space
-complexity due to using the already present [NX](#no-executeexecute-neverexecute-disable-attribute-nxxnxd) bit.
-
-### Available Settings
-
-- EfiReservedMemoryType
-- EfiLoaderCode - If an image does not indicate support for
-[NX](#no-executeexecute-neverexecute-disable-attribute-nxxnxd) via the [NXCOMPAT](#nxcompat) DLL flag, the logic will
-cease to set the [NX](#no-executeexecute-neverexecute-disable-attribute-nxxnxd) attribute on allocations of memory
-this type. Using the BlockImagesWithoutNxFlag in the [Image Protection Policy](image-protection-policy) will prevent
-images which don't support [NXCOMPAT](#nxcompat).
-- EfiLoaderData
-- EfiBootServicesCode - Same note as EfiLoaderCode.
-- EfiBootServicesData
-- EfiRuntimeServicesCode - Same note as EfiLoaderCode.
-- EfiRuntimeServicesData
-- EfiConventionalMemory
-- EfiUnusableMemory
-- EfiACPIReclaimMemory
-- EfiACPIMemoryNVS
-- EfiMemoryMappedIO
-- EfiMemoryMappedIOPortSpace
-- EfiPalCode
-- EfiPersistentMemory
-- OEMReserved
-- OSReserved
-
-## **Page Guards**
-
-### Summary
-
-The HeapGuardPageType policy implements guard pages on the specified memory types to detect heap overflow. If a bit
-is set, a guard page will be added before and after the
-corresponding type of page allocated if there's enough free pages for all of them. Guard pages are set to NOT PRESENT so any attempt
-to access them will cause a [#PF](#page-fault-exception-pf). The system will do its best to ensure that only one guard page separates two allocated pages to avoid wasted space.
-
-### Overhead
-**O(n)** time where n is the number of page allocations/frees. Because there are 2 extra pages allocated for
-every call to AllocatePages(), **O(n)** space is also required.
-
-### Available Settings for DXE and MM
-
-- EfiReservedMemoryType
-- EfiLoaderCode
-- EfiLoaderData
-- EfiBootServicesCode
-- EfiBootServicesData
-- EfiRuntimeServicesCode
-- EfiRuntimeServicesData
-- EfiConventionalMemory
-- EfiUnusableMemory
-- EfiACPIReclaimMemory
-- EfiACPIMemoryNVS
-- EfiMemoryMappedIO
-- EfiMemoryMappedIOPortSpace
-- EfiPalCode
-- EfiPersistentMemory
-- OEMReserved
-- OSReserved
-
-## **Pool Guards**
-
-### Summary
+- **UefiNullDetection**: Enable NULL pointer detection for DXE.
+- **DisableEndOfDxe**: Disable NULL pointer detection just after the end of DXE protocol
+is installed.
+- **DisableReadyToBoot**: Disable NULL pointer detection just after the ready to boot
+protocol is installed.
+
+The **MM environment** only has a single option indicating whether NULL detection
+is active or not.

```

Comment:
I think a little more description of why DisableEndOfDxe and DisableReadyToBoot exist would be useful (i.e. historical problems seen or why a bootloader needs it so, etc.).

Reply:
Added an extra sentence to describe why these two policy options are available.